src/Tracking/Events/ActivatedEvents.php

```diff
@@ -50,6 +58,9 @@ public function register(): void {}
 	 * Track when the extension is activated from a source.
 	 */
 	public function maybe_track_activation_source(): void {
+		// Update db flag.
+		$this->options->update( OptionsInterface::TRACKED_ACTIVATION_SOURCE, true );
```

Comment:
If our intention is to have an option keep track of whether the `activate` code was run, then I don't think this option is needed, since we can confirm that the main `activate` function is called only once.